### PR TITLE
Add http_forwarder extension

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Add a `http_forwarder` extension that forwards HTTP requests to a specified target.

**Testing:** Tests added.

**Documentation:** README added.

### CODE REVIEW

File: extension/httpforwarder/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+* `headers`: Additional headers to be added to all requests passing through the extension.
+* `http_timeout` (default: `10s`): How long to wait for each request to complete.
+
+### Example
+
+```yaml
+  http_forwarder:
+    forward_to: http://target/
+    headers:
+      otel_http_forwarder: dev
```
Comment (jrcamp) [inline extension/httpforwarder/README.md:23]: Could this use HTTPClientSettings? Config would look something like:

```yaml
  http_forwarder:
    upstream:
      endpoint: http://target
      tls_settings:
             ...
      headers:
```

Comment (asuresh4) [reply extension/httpforwarder/README.md]: Updated to make use of `confighttp.HTTPClientSettings` for client options. Let me know what you think.

File: extension/httpforwarder/extension.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	server := &http.Server{
+		Addr:    config.Endpoint,
+		Handler: handler,
+	}
+
+	h := &httpForwarder{
+		listenAt:  config.Endpoint,
+		forwardTo: url,
+		headers:   config.Upstream.Headers,
+		httpClient: &http.Client{
```
Comment (jrcamp) [inline extension/httpforwarder/extension.go:105]: Use config.upstream.ToClient()

**Comment by bogdandrutu** [comment]
@asuresh4 can I use you as a validator for our proposal on how to add new components (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/977) to validate that the workflow works?

**Comment by asuresh4** [comment]
> @asuresh4 can I use you as a validator for our proposal on how to add new components (#977) to validate that the workflow works?

@bogdandrutu - open #979 as a first PR instead as suggested in your proposal. Will close this one out.

File: extension/httpforwarder/extension.go (hunk 2 above)
Comment (asuresh4) [reply extension/httpforwarder/extension.go:105]: Addressed in #979

**Pull request closed without merging**